pkg/envoy/ads/cache_callbacks.go

```diff
@@ -70,15 +74,33 @@ func (s *Server) OnStreamOpen(ctx context.Context, streamID int64, typ string) e
 	return nil
 }
 
-func (s *Server) update(proxy *envoy.Proxy) {
-	ch := s.workqueues.AddJob(&proxyResponseJob{
-		proxy:     proxy,
-		xdsServer: s,
-		typeURIs:  envoy.XDSResponseOrder,
-		done:      make(chan struct{}),
-	})
-	<-ch
-	close(ch)
+func (s *Server) scheduleUpdate(proxy *envoy.Proxy) {
+	var wg sync.WaitGroup
+	wg.Add(1)
+	s.workqueues.AddJob(
+		func() {
+			t := time.Now()
+			log.Debug().Msgf("Starting update for proxy %s", proxy.String())
+
+			if err := s.update(proxy); err != nil {
+				log.Error().Err(err).Str("proxy", proxy.String()).Msg("Error generating resources for proxy")
+			}
+			log.Debug().Msgf("Update for proxy %s took took %v", proxy.String(), time.Since(t))
+			wg.Done()
+		})
+	wg.Wait()
```

Comment:
could this cause some delays if there are other things in the queue and we are waiting for it complete?

Reply:
ya but it's somewhat meaningless to schedule multiple simultaneously. each schedule tries to grab a new state, so we don't know if the current job has all the state or not, or whether the new one will anyways.

We use "coalescing" in the message broker to account for this. It probably makes sense to move that coalescing here, but I'll leav that out of this PR for now.

This mimics the current setup as well.